### PR TITLE
Improve navigation cpp

### LINKED ISSUE #946: Feature: remote control support for Android TV
* need to add up and down key handling (probably also need to add right and left key handling)
* need to add "focus effect" display on controls for android

### LINKED ISSUE #1001: remote control not working on android tv version 4.7.0.0
tv remote control not working on new version on android tv
I have to use mouse to be able to turn vpn on or off

tv is tcl with android version 11
app is 4.7.0.0 manually installed apk from github

### PR DESCRIPTION
**Pull request by CyAn84**
This is still draft pull request since the transitions need to be fixed.
The idea of navigation is follow:
there are QML components which should be focused with property `isFocusable` with key Tab handlers, `focusController` goes through all visible components with this property.
No need to describe the focus logic manually but the focus navigation algorythm is straightforward - from upper left to bottom right.

### CODE REVIEW

File: client/ui/qml/Components/HomeSplitTunnelingDrawer.qml [hunk 1]
```diff
@@ -59,11 +46,9 @@ DrawerType2 {
             descriptionText: qsTr("Enabled \nCan't be disabled for current server")
             rightImageSource: "qrc:/images/controls/chevron-right.svg"
 
-            KeyNavigation.tab: siteBasedSplitTunnelingSwitch.visible ? siteBasedSplitTunnelingSwitch.rightButton : focusItem
-
             clickedFunction: function() {
-//                PageController.goToPage(PageEnum.PageSettingsSplitTunneling)
-//                root.close()
+               PageController.goToPage(PageEnum.PageSettingsSplitTunneling)
```
Comment (Nethius) [inline client/ui/qml/Components/HomeSplitTunnelingDrawer.qml:50]: please, return the commented line

File: client/ui/qml/Pages2/PageSettings.qml [hunk 2]
```diff
@@ -156,10 +140,8 @@ PageType {
                 Layout.preferredHeight: about.height
 
                 text: qsTr("Close application")
+                rightImageSource: "qrc:/images/controls/chevron-right.svg"
```
Comment (Nethius) [inline client/ui/qml/Pages2/PageSettings.qml]: please, remove this

File: client/ui/qml/Pages2/PageSettingsSplitTunneling.qml [hunk 3]
```diff
@@ -179,124 +169,120 @@ PageType {
...
-        id: fl
+    ListView {
+        id: sites
+
         anchors.top: header.bottom
         anchors.topMargin: 16
-        contentHeight: col.implicitHeight + addSiteButton.implicitHeight + addSiteButton.anchors.bottomMargin + addSiteButton.anchors.topMargin
+        width: parent.width
+
+        height: 200 // TODO: Change to correct height
```
Comment (aiamnezia) [inline client/ui/qml/Pages2/PageSettingsSplitTunneling.qml]:
```suggestion
        anchors.bottom: addSiteButton.top
```

File: client/ui/qml/Pages2/PageSettingsSplitTunneling.qml [hunk 4]
```diff
@@ -179,124 +169,120 @@ PageType {
...
 
         enabled: root.pageEnabled
 
-        Column {
-            id: col
-            anchors.top: parent.top
-            anchors.left: parent.left
-            anchors.right: parent.right
+        property bool isFocusable: true
```
Comment (aiamnezia) [inline client/ui/qml/Pages2/PageSettingsSplitTunneling.qml:186]:
```suggestion

        ScrollBar.vertical: ScrollBar {}
```

Comment (aiamnezia) [reply client/ui/qml/Pages2/PageSettingsSplitTunneling.qml:186]: The default behavior of the ScrollBar is not working correctly. To fix this, you can initialize the ScrollBar as an empty object. Qt will then set the necessary default values automatically.

File: client/ui/qml/Pages2/PageSettingsSplitTunneling.qml [hunk 5]
```diff
@@ -179,124 +169,120 @@ PageType {
...
+        id: sites
+
         anchors.top: header.bottom
         anchors.topMargin: 16
-        contentHeight: col.implicitHeight + addSiteButton.implicitHeight + addSiteButton.anchors.bottomMargin + addSiteButton.anchors.topMargin
+        width: parent.width
+
+        height: 200 // TODO: Change to correct height
 
         enabled: root.pageEnabled
```
Comment (aiamnezia) [inline client/ui/qml/Pages2/PageSettingsSplitTunneling.qml:183]:
```suggestion
        enabled: root.pageEnabled
        reuseItems: true
```

for optimization

File: client/ui/qml/Components/SelectLanguageDrawer.qml [hunk 6]
```diff
@@ -43,167 +30,151 @@ DrawerType2 {
 
             BackButtonType {
                 id: backButton
+
+                Layout.fillWidth: true
+                Layout.topMargin: 16
```
Comment (Nethius) [inline client/ui/qml/Components/SelectLanguageDrawer.qml]: There's no need for margins

File: client/ui/qml/Pages2/PageSettingsSplitTunneling.qml (hunk 5 above)
Comment (CyAn84) [reply client/ui/qml/Pages2/PageSettingsSplitTunneling.qml:183]: Done

File: client/ui/qml/Pages2/PageSettingsSplitTunneling.qml (hunk 4 above)
Comment (CyAn84) [reply client/ui/qml/Pages2/PageSettingsSplitTunneling.qml:186]: I added ScrollBarType component and use it on all ListViews

File: client/ui/qml/Pages2/PageSetupWizardConfigSource.qml [hunk 7]
```diff
@@ -187,143 +184,129 @@ PageType {
...
-                        }
-                    }
-                }
+    QtObject {
+        id: backupRestore
+
+        property string title: qsTr("Restore from backup")
+        property string description: qsTr("")
+        property string imageSource: "qrc:/images/controls/archive-restore.svg"
+        property bool isVisible: true
```
Comment (Nethius) [inline client/ui/qml/Pages2/PageSetupWizardConfigSource.qml]: visible: PageController.isStartPageVisible()

File: client/ui/qml/Pages2/PageSetupWizardConfigSource.qml [hunk 8]
```diff
@@ -187,143 +184,129 @@ PageType {
...
+        id: siteLink
 
-                onClicked: {
-                    Qt.openUrlExternally(LanguageModel.getCurrentSiteUrl())
-                }
-            }
+        property string title: qsTr("I have nothing")
+        property string description: qsTr("")
+        property string imageSource: "qrc:/images/controls/help-circle.svg"
+        property bool isVisible: true
```
Comment (Nethius) [inline client/ui/qml/Pages2/PageSetupWizardConfigSource.qml]: visible: PageController.isStartPageVisible()

File: client/ui/qml/Pages2/PageSetupWizardConfigSource.qml (hunk 7 above)
Comment (CyAn84) [reply client/ui/qml/Pages2/PageSetupWizardConfigSource.qml]: Done

File: client/ui/qml/Pages2/PageSetupWizardConfigSource.qml (hunk 8 above)
Comment (CyAn84) [reply client/ui/qml/Pages2/PageSetupWizardConfigSource.qml]: Done

File: client/ui/qml/Components/SelectLanguageDrawer.qml (hunk 6 above)
Comment (CyAn84) [reply client/ui/qml/Components/SelectLanguageDrawer.qml]: Done

**Review by Nethius — APPROVED** [review]

**Pull request merged**